source/serverless-testing/README.rst

```diff
@@ -0,0 +1,125 @@
+======================
+Atlas Serverless Tests
+======================
+
+.. contents::
+
+----
+
+Introduction
+============
+
+This file describes a subset of existing tests that drivers MUST use to assert
+compatibility with Atlas Serverless.
+
+Serverless Configuration
+========================
+
+These tests MUST be run against a live Atlas Serverless instance. A new instance
+MUST be created each time the test suite is run, and that instance MUST be used
+for all of the tests required by this specification. Once the tests are
+finished, the instance MUST be deleted regardless of the outcome of the tests.
+The `serverless directory in the drivers-evergreen-tools repository`_ contains
+scripts for creating and deleting Atlas Serverless instances, and the
+``config.yml`` contains an example Evergreen configuration that uses them to run
+the tests. It can take up to 15 minutes or so to provision a new Atlas
+Serverless instance, so it is recommended to create one manually via the scripts
+in drivers-evergreen-tools that can be reused for the initial implementation of
+the tests before moving to Evergreen patches.
+
+Drivers MUST use the `create-instance.sh`_ script in ``drivers-evergreen-tools``
+to create a new Atlas Serverless instance for Evergreen tasks. The script writes
+two URIs to a YAML expansions file:
+
+- ``MULTI_ATLASPROXY_SERVERLESS_URI`` A URI to the load balancer fronting
+  multiple Atlas Proxy processes.
+
+- ``SINGLE_ATLASPROXY_SERVERLESS_URI`` A URI to one of the Atlas Proxy
+  processes.
+
+The URIs can be read by drivers via the ``expansions.update`` Evergreen command.
+This will store the URIs into the ``SINGLE_ATLASPROXY_SERVERLESS_URI`` and
+``MULTI_ATLASPROXY_SERVERLESS_URI`` Evergreen expansions.
+
+.. _serverless directory in the drivers-evergreen-tools repository: https://github.com/mongodb-labs/drivers-evergreen-tools/tree/1ca6209825b6ed07ce90e24cda659143443709c8/.evergreen/serverless
+
+All tests MUST be run with wire protocol compression and authentication
+enabled.
+
+Test Runner Configuration
```

Comment:
I noted that these URIs to satisfy load balancer URIs in test runners and that they are analogous to `SINGLE_LB_MONGOS_URI` and `MULTI_LB_MONGOS_URI`.

I considered mentioning these variables in the Retryable Reads/Writes or Unified Test Format instructions, but I chose not to. Those instructions do not mention `SINGLE_LB_MONGOS_URI` or `MULTI_LB_MONGOS_URI` specifically, so I do not think those test runners need to treat these any differently.

I considered reusing the variable names `SINGLE_LB_MONGOS_URI` and `MULTI_LB_MONGOS_URI`. Since mongos is not part of serverless, I chose not to.